src/lib/pages/module-details/index.tsx

```diff
@@ -38,16 +49,27 @@ const ModuleDetailsBody = ({
   address,
   moduleName,
   tab,
-}: ModuleDetailsQueryParams) => {
+}: ModuleDetailsQueryParams | ModuleDetailsLiteQueryParams) => {
   const router = useRouter();
   const navigate = useInternalNavigate();
   const formatAddresses = useFormatAddresses();
   const { hex: vmAddress } = formatAddresses(address);
 
-  const { data, isLoading } = useModuleData(vmAddress, moduleName);
+  const isFullTier = useTierConfig() === "full";
```

Comment:
what about manually handle lite tabIndex here, so we don't have to create `TabIndexLite` and `zModuleDetailsLiteQueryParams`?

e.g.
```
const currentTab = !isFullTier && tab === TabIndex.TxsHistories ? TabIndex.Overview : tab;
```

up to debate @Poafs1 @evilpeach